exercises/practice/book-store/.meta/example.rb

```diff
@@ -1,46 +1,29 @@
+require 'set'
+
 class BookStore
 
-  GROUP_DISCOUNTS = [0, 0.05, 0.1, 0.2, 0.25]
+  GROUP_DISCOUNTS = [1.0, 0.95, 0.9, 0.8, 0.75]
   INDIVIDUAL_PRICE = 8
 
   def self.calculate_price(basket)
-    #FIXME: find a real solution instead of hardcoding :(
-    return 51.20 if basket == [1, 1, 2, 3, 4, 4, 5, 5]
+    return 0.00 if basket.empty?
 
     groups = []
-    remaining_books = basket.dup
-
-    # Make as many groups of 4 as you can
-    while (largest_group = remaining_books.uniq).length > 3
-      group_of_four = largest_group[0..3]
-      group_of_four.each {|book| remaining_books.delete_at(remaining_books.index(book)) }
-      groups << group_of_four
-    end
-
-    # Bump as many of them to groups of 5 as you can
-    groups.each do |group|
-      fifth_book = (remaining_books - group).first
-      next unless fifth_book
-
-      group << fifth_book
-      remaining_books.delete_at(remaining_books.index(fifth_book))
-    end
 
     # Make the largest groups you can with the remaining books
-    while (new_group = remaining_books.uniq).any?
-      new_group.each {|book| remaining_books.delete_at(remaining_books.index(book)) }
-      groups << new_group
+    while !basket.empty?
+      set = Set.new(basket)
+      groups << set.size
+      set.each{|book| basket.delete_at(basket.index(book))} # Remove books in set from basket
```

Comment:
I do tend to be a little overzealous when commenting my code, so you're probably right about it being surplus to requirements.  As for the use of a set, I was using it as a quick way of gathering all the unique elements from the basket together into a group (I didn't realise that #uniq was already there and ready to be used... oops!).

But since (upon closer examination) this loop is doing almost precisely what the old loop was doing, I'm thinking of just deleting my one and sticking with the old loop.  While I think that my version 'looks' better (notwithstanding the set) I do appreciate the cool trick (in the old loop) of assigning the new_group variable and then testing it all in the one line (I didn't know that was a thing I could do, so I'm very tempted to keep it in).